### ISSUE TITLE
Call to undefined method Spatie\MediaLibrary\UrlGenerator\S3UrlGenerator::getPath

### ISSUE THREAD

**Issue by sebestenyb** [spatie/laravel-medialibrary#534]
Having this issue on v5:

```
[Symfony\Component\Debug\Exception\FatalThrowableError]
Call to undefined method Spatie\MediaLibrary\UrlGenerator\S3UrlGenerator::getPath()

Exception trace:
 () at /vendor/spatie/laravel-medialibrary/src/Media.php:81
 Spatie\MediaLibrary\Media->getPath() at /vendor/spatie/laravel-medialibrary/src/ImageGenerators/BaseGenerator.php:24
 Spatie\MediaLibrary\ImageGenerators\BaseGenerator->canConvert() at /vendor/spatie/laravel-medialibrary/src/FileManipulator.php:120
```

A blank new Laravel 5.4 project.

```    
// composer.json
"require": {
        "php": ">=5.6.4",
        "laravel/framework": "5.4.*",
        "laravel/tinker": "~1.0",
        "league/flysystem-aws-s3-v3": "^1.0",
        "spatie/laravel-medialibrary": "^5.0"
    },
```

```
<?php
// User.php
namespace App;

use Illuminate\Notifications\Notifiable;
use Illuminate\Foundation\Auth\User as Authenticatable;
use Spatie\MediaLibrary\HasMedia\HasMediaTrait;
use Spatie\MediaLibrary\HasMedia\Interfaces\HasMedia;

class User extends Authenticatable implements HasMedia
{
    use Notifiable, HasMediaTrait;
```

```
// filesystem.php
        'media' => [
            'driver' => 's3',
            'key' => env('AWS_KEY'),
            'secret' => env('AWS_SECRET'),
            'region' => env('AWS_REGION'),
            'bucket' => env('AWS_BUCKET'),
        ],    
```

```
// tinker

Psy Shell v0.8.1 (PHP 7.0.8 — cli) by Justin Hileman
>>> use App\User
=> null
>>> use Faker\Factory
=> null
>>> $faker = Factory::create()
=> Faker\Generator {#678}
>>> User::first()->addMediaFromUrl($faker->imageUrl())->toMediaLibrary()
PHP error:  Call to undefined method Spatie\MediaLibrary\UrlGenerator\S3UrlGenerator::getPath() in /Users/balazssebesteny/Sites/medialibrary-test/vendor/spatie/laravel-medialibrary/src/Media.php on line 81
```

**Comment by Mirdrack** [comment]
I have the same problem here but only with files that are not images
If I upload an image everything is right but if I try with a text file I get an error
Is there a way to filter by file type?

**Comment by njbarrett** [comment]
This is also an issue for me.
Laravel 5.4 / "spatie/laravel-medialibrary": "^5.0.0",

**Comment by freekmurze** [comment]
The fix by @AlexVanderbist is now tagged as `5.3.1`

**Comment by sebestenyb** [comment]
#546
